### PR TITLE
refactor: All aboard the jest train

### PR DESCRIPTION
**Pull request by ntwb**
aka Switch from AVA to Jest

### CODE REVIEW

**Comment by m-allanson** [comment]
@ntwb nice 👍 

I think this suffers from the same issue with preserving descriptions that @davidtheclark mentioned over at https://github.com/stylelint/stylelint/pull/2085#pullrequestreview-9354034

An updated example for the valid css tests might look like this?

```js
describe("flags no warnings with valid css", () => {
  let result

  beforeEach(() => {
    result = stylelint.lint({
      code: validCss,
      config: config,
    })
  })

  it("did not error", () => {
    return result.then(data => (
      expect(data.errored).toBeFalsy()
    ))
  })

  it("flags no warnings", () => {
    return result.then(data => (
      expect(data.results[0].warnings.length).toBe(0)
    ))
  })
})
```

File: __tests__/index.js [hunk 1]
```diff
@@ -82,29 +81,29 @@ const invalidCss = (
 
 `)
 
-test("no warnings with valid css", t => {
+it("Shows no warnings with valid css", () => {
```
Comment (jeddy3) [inline __tests__/index.js]: Lets go with a lowercase "s" for consistency with tests in the main repo.

File: package.json [hunk 2]
```diff
@@ -32,13 +34,14 @@
     "stylelint": "^7.6.0"
   },
   "scripts": {
-    "ava": "ava --verbose \"__tests__/**/*.js\"",
+    "jest": "jest",
```
Comment (jeddy3) [inline package.json]: I think this is redundant now and can go

File: package.json [hunk 3]
```diff
@@ -32,13 +34,14 @@
...
     "lint:js": "eslint . --ignore-path .gitignore",
     "lint:md": "remark . --quiet --frail",
     "lint": "npm-run-all --parallel lint:*",
     "pretest": "npm run lint",
     "release": "npmpub",
-    "test": "npm run ava"
+    "test": "npm run jest",
+    "watch": "jest --watch"
```
Comment (jeddy3) [inline package.json]: 👍

**Comment by jeddy3** [comment]
Nice! In addition to @m-allanson suggestions, I've made a minor two.

File: package.json (hunk 2 above)
Comment (ntwb) [inline package.json]: Done in https://github.com/stylelint/stylelint-config-standard/pull/50/commits/f73e8e8bdf787749bbbdf54ed96a6ec05a9da65b (I originally had cache issues, and was using `jest --no-cache` here)

File: __tests__/index.js (hunk 1 above)
Comment (ntwb) [inline __tests__/index.js]: To this is lame, I change `S` to `s` and the [tests fail](https://gist.github.com/ntwb/5b845b34f450c8ea869c616acbb060ae) /shrug

**Comment by ntwb** [comment]
Thanks @m-allanson, thats working perfectly for the valid css case 👍 

Here's where I've got to for the invalid css test:
```js
describe("flags warnings with invalid css", () => {
  let result

  beforeEach(() => {
    result = stylelint.lint({
      code: invalidCss,
      config: config,
    })
  })

  it("did error", () => {
    return result.then(data => (
      expect(data.errored).toBeTruthy()
    ))
  })

  it("flags one warning", () => {
    return result.then(data => (
      expect(data.results[0].warnings.length).toBe(1)
    ))
  })

  it("correct warning text", () => {
    return result.then(data => (
      expect(data.results[0].warnings.text).toBe("Expected a leading zero (number-leading-zero)")
    ))
  })
})
```

That results in 2 out of 3 passing:
```shell
  flags warnings with invalid css
    ✓ did error (11ms)
    ✓ flags one warning (10ms)
    ✕ correct warning text (7ms)
```

The 3rd and failing test displays this error:
```shell
 ● flags warnings with invalid css › correct warning text

    expect(received).toBe(expected)
    
    Expected value to be (using ===):
      "Expected a leading zero (number-leading-zero)"
    Received:
      undefined
    
    Difference:
    
    Comparing two different types of values:
      Expected: string
      Received: undefined
      
      at __tests__/index.js:131:47
```

I've been trying to work out the cause of the above for the past hour and now I'm out of ideas, and out time for today, any tips on what I'm missing here would be greatly appreciated 😄

**Comment by davidtheclark** [comment]
@ntwb `data.results[0].warnings` is an array, so it has no `text` property. Probably you want `data.results[0].warnings[0].text`?

**Comment by ntwb** [comment]
Thanks @davidtheclark 😃 

Is it worth adding the following tests to test against the returned results also for invalid CSS?

Results:
```json
    "warnings": [{
        "line": 2,
        "column": 8,
        "rule": "number-leading-zero",
        "severity": "error",
        "text": "Expected a leading zero (number-leading-zero)"
    }]
```

Tests:
```js
  it("correct rule flagged", () => {
    return result.then(data => (
      expect(data.results[0].warnings[0].rule).toBe("number-leading-zero")
    ))
  })

  it("correct severity flagged", () => {
    return result.then(data => (
      expect(data.results[0].warnings[0].severity).toBe("error")
    ))
  })

  it("correct line number", () => {
    return result.then(data => (
      expect(data.results[0].warnings[0].line).toBe(2)
    ))
  })

  it("correct column number", () => {
    return result.then(data => (
      expect(data.results[0].warnings[0].column).toBe(8)
    ))
  })
```

**Comment by jeddy3** [comment]
> Is it worth adding the following tests to test against the returned results also for invalid CSS?

I think so. Anything to give us confidence that the right error _at the right place_ is being returned is a good thing.

**Comment by ntwb** [comment]
> I think so. Anything to give us confidence that the right error at the right place is being returned is a good thing.

Done, added in https://github.com/stylelint/stylelint-config-standard/pull/50/commits/b27da27b6f4836e443f6ea0ed1061fa32a93ca4d

**Comment by jeddy3** [comment]
@ntwb Thanks! I wanted to see what happened with the dropping babel and migrating to jest before merging this. Things seem to be pointing in one direction now, so merging :)

**Comment by ntwb** [comment]
Thanks everyone 🚆

**Pull request merged**